rust/sbp/src/de.rs

```diff
@@ -206,6 +278,59 @@ mod tests {
 
     use super::*;
 
+    struct ZeroReader;
+
+    impl std::io::Read for ZeroReader {
+        fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+            Ok(buf.len())
+        }
+    }
```

Comment:
Should probably be 

```rust
    impl std::io::Read for ZeroReader {
        fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
            Ok(0)
        }
    }
```

if buf had some bytes in it before for some reason, this read impl would say it read some bytes even though it didn't

Reply:
Your suggested reader actually gets handled currently. What I had originally was simply just returning the buffer assuming it was filled with zeros. This was just abusing the same corner case in the other testcase, so per your suggestion I have merged the two. I decided to keep the "NeverPreambleReader" as I think it describes the corner case better than ZeroReader unless you have a better suggestion?

Reply:
Ahhh yeah that makes sense, sounds good to me